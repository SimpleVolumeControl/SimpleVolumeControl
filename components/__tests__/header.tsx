import { render } from '@testing-library/preact';
import Header from '../header';
import { RecoilRoot } from 'recoil';

describe('Header', () => {
  test('should render correctly on home page', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <Header isHome={true} />
      </RecoilRoot>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render correctly on other pages', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <Header isHome={false} />
      </RecoilRoot>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('should hide logout button when no password is set', () => {
    const { asFragment } = render(
      <RecoilRoot>
        <Header isHome={false} />
      </RecoilRoot>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test('should show logout button when password is set', () => {
    // TODO
  });
});
