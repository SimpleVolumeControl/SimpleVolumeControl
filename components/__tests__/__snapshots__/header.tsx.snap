// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should hide logout button when no password is set 1`] = `
<DocumentFragment>
  <div
    class="navbar mb-4 shadow-lg bg-primary text-neutral-content rounded-box flex-wrap xs:flex-nowrap"
  >
    <div
      class="navbar-start"
    >
      <a
        class="btn btn-outline btn-accent"
        href="/"
      >
        zurück
      </a>
    </div>
    <div
      class="px-2 mx-2 navbar-center w-full xs:w-auto order-first xs:order-none justify-center"
    >
      <span
        class="text-lg sm:text-3xl sm:font-thin"
      >
        SimpleVolumeControl
      </span>
    </div>
    <div
      class="navbar-end"
    />
  </div>
</DocumentFragment>
`;

exports[`Header should render correctly on home page 1`] = `
<DocumentFragment>
  <div
    class="navbar mb-4 shadow-lg bg-primary text-neutral-content rounded-box flex-wrap xs:flex-nowrap"
  >
    <div
      class="navbar-start"
    />
    <div
      class="px-2 mx-2 navbar-center w-full xs:w-auto order-first xs:order-none justify-center"
    >
      <span
        class="text-lg sm:text-3xl sm:font-thin"
      >
        SimpleVolumeControl
      </span>
    </div>
    <div
      class="navbar-end"
    />
  </div>
</DocumentFragment>
`;

exports[`Header should render correctly on other pages 1`] = `
<DocumentFragment>
  <div
    class="navbar mb-4 shadow-lg bg-primary text-neutral-content rounded-box flex-wrap xs:flex-nowrap"
  >
    <div
      class="navbar-start"
    >
      <a
        class="btn btn-outline btn-accent"
        href="/"
      >
        zurück
      </a>
    </div>
    <div
      class="px-2 mx-2 navbar-center w-full xs:w-auto order-first xs:order-none justify-center"
    >
      <span
        class="text-lg sm:text-3xl sm:font-thin"
      >
        SimpleVolumeControl
      </span>
    </div>
    <div
      class="navbar-end"
    />
  </div>
</DocumentFragment>
`;
